### ISSUE TITLE
Represent geo coordinates as two element arrays.

### ISSUE THREAD

**Issue by WardCunningham** [clemsos/wiki-plugin-map#9]
I chose to represent markers as:

```
{lat: nn.nnn, lon: nnn.nnn, label: "xxxxx"}
```

I suggest we change that format to:

```
{latLng: [nn.nnnn, nnn.nnn], label: "xxxxx"}
```

My original thought was that a convention based on two order-dependent numbers could be confused with the [x,y] convention. That doesn't seem to be the case as [y,x] is common  and we are often converting to that. It helps that `latLng` reminds us that we are using a cartographic, not geometric, convention. Note I also suggest converting from `lon` to `Lng` which also appears to be convention.

We would need to coordinate this change with @opn who may be generating the original format. I am against accepting either format as that would surely increase confusion.

**Comment by opn** [comment]
Was the idea not to use standard GeoJSON
<http://plugin.fedwiki.org/geojson.html>?

{ "type": "Feature",
        "geometry": {"type": "Point", "coordinates": [102.0, 0.5]},
        "properties": {"prop0": "value0"}
        },


It makes sense to expose a javascript object in our page-json that conforms
to this specification. We then get a nice smooth upgrade path as we want to
add topology - etc?

**Comment by WardCunningham** [comment]
Ok. Will need to say what type we expect and what properties we interpret. Good time to agree on this. 

Best regards -- Ward

**Comment by WardCunningham** [comment]
Can we use type MultiFeature? That would handle all our cases. 

Best regards -- Ward

**Comment by WardCunningham** [comment]
After consulting the geojson spec I will suggest that the subset we exchange between items be the javascript objects that would be the parsed versions of this geojson. 

Empty collection of markers:

```
{
	"type": "FeatureCollection",
	"features": []
}
```

A collection with one marker:

```
{
	"type": "FeatureCollection",
	"features": [{
		"type": "Feature",
		"geometry": {
			"type": "Point",
			"coordinates": [102.0, 0.5]
		},
		"properties": {
			"label": "Wiki with [[Link]]"
		}
	}]
}
```

These fields must be present. Any fields other than these will be ignored.

This is the interchange format between plugins using the `map-source` and `getGeo` discovery protocol. The map plugin is free to use whatever representations are convenient inside the plugin's javascript. These are free to change at any time.

Note: any internal link found in the label properties cannot be reliably interpreted independent of wiki resolution-context provided by a wiki page. See [Collaborative Link](http://ward.bay.wiki.org/collaborative-link.html)

**Comment by WardCunningham** [comment]
Section [2.1.1](http://geojson.org/geojson-spec.html#positions) explains that Point coordinates are of the form `[x, y]`, not `[y, x]` as I was proposing above. Considering this, I suggest we preserve the existing internal format `{lat: y, lon: x}` so that an api-appropriate order can be constructed at each call.

**Comment by opn** [comment]
Seems good to me.

**Comment by clemsos** [comment]
Agree, we should keep the more literal and explicit form of `{lat: y, lon: x}` . For a bit of background on this, this [blog post](http://www.macwright.org/lonlat/) shows how inconsistent the `[lat, lon]` or `[lon, lat]` ordering is in the geospatial data world
